app/routes.py

```diff
@@ -1,2 +1,272 @@
+            return jsonify ({
+                "details": "Invalid data"
+            }), 400
+
+
+        new_task = Task(
+            title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"]
+        )
+        db.session.add(new_task)
+        db.session.commit()
+        
+#Wave 1: Create A Task: Valid Task with null completed_at 201 CREATED
+
+
+        return jsonify({"task":new_task.to_dict()}),201
+# handle_one_task handles GET,PUT and DELETE requests for the tasks/task_id endpoint
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_one_task(task_id):
+    valid_int(task_id,"task_id")
+    task = Task.query.get_or_404(task_id)
+# Wave 1: Get One Task: One Saved Task
+    if request.method == "GET":
+        has_complete = task.completed_at
+        if task.goal_id:
+            task_response={   
+                    "task": {
+                    "id": task.id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if has_complete == None else has_complete,
+                    
+                }
+                }
+        else:
+            task_response={   
+                    "task": {
+                    "id": task.id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if has_complete == None else has_complete,
+                    
+                }
+                }
+        
+        return jsonify(task_response)
+#Wave 1: Update Task, #Wave 1 Update Task: No Matching Task, Update Task 200 OK
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        
+
+        db.session.commit()
+        return jsonify({"task":task.to_dict()}),200
+
+#Wave 1  Delete Task: Deleting A Task, #Wave 1: Delete Task: No Matching Task
+    elif request.method == "DELETE": 
+        db.session.delete(task)
+        db.session.commit()
+        response = {
+            "details": f'Task {task.id} "{task.title}" successfully deleted'
+        }
+        json_response = jsonify(response)
+        return make_response(json_response, 200)
+
+def slack_bot(title):
+    query_path = {
+        "channel": "melinda-bot",
+        "text": f"Someone completed the task {title}"
+    }
+    header = {
+        "Authorization": f"Bearer {os.environ.get('BOT')}"
+    }
+    response = requests.post("https://slack.com/api/chat.postMessage",params = query_path, headers = header)
+    return response.json()
+
+
+#Wave 3 
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_completed_task(task_id):
```

Comment:
Great work, the use of helper functions and get_or_404 in this function really streamline the code!